packages/wrangler/src/vectorize/insert.ts

```diff
@@ -30,13 +41,32 @@ export async function handler(
 	args: StrictYargsOptionsToInterface<typeof options>
 ) {
 	const config = readConfig(args.config, args);
+	const file = await open(args.file);
 
-	const vectors: VectorizeVector[] = [];
-	if (args.vectors) {
-		// Parse each vector into a Vector type
-		// Think about potential limits on args.vectors.length?
+	let index: Optional<VectorizeVectorMutation, "ids"> | undefined;
+	for await (const batch of getBatchFromFile(file, args.batchSize)) {
+		const formData = new FormData();
+		formData.append("vectors", new File([batch.join(`\n`)], "vectors.ndjson"));
+		logger.log(`✨ Uploading vector batch (${batch.length} vectors)`);
+		const idxPart = await insertIntoIndex(config, args.name, formData);
+		if (!index) index = idxPart;
+		else index.count += idxPart.count;
 	}
 
-	const index = await insertIntoIndex(config, args.name, vectors);
+	// remove the ids - skip tracking these for bulk uploads since this could be in the 100s of thousands.
+	if (index) delete index.ids;
 	logger.log(JSON.stringify(index, null, 2));
 }
+
+// helper method that reads an ndjson file line by line in batches. not this doesn't
+// actually do any parsing - that will be handled on the backend
+async function* getBatchFromFile(file: FileHandle, batchSize = 3) {
+	let batch: string[] = [];
+	for await (const line of file.readLines()) {
+		if (batch.push(line) >= batchSize) {
```

Comment:
Note this does no validation here: it just takes the line as it is. We'll need to do verification on the backend anyway, so this should be okay. Also it appears there's no `zod` anywhere in wrangler evidently most folks don't schema validate client side here